ceph.spec.in

```diff
@@ -1332,7 +1332,7 @@ chmod 0644 %{buildroot}%{_docdir}/ceph/sample.ceph.conf
 install -m 0644 -D COPYING %{buildroot}%{_docdir}/ceph/COPYING
 install -m 0644 -D etc/sysctl/90-ceph-osd.conf %{buildroot}%{_sysctldir}/90-ceph-osd.conf
 
-install -m 0755 src/ceph-daemon %{buildroot}%{_sbindir}/ceph-daemon
+install -m 0755 src/ceph-daemon/ceph-daemon %{buildroot}%{_sbindir}/ceph-daemon
```

Comment:
Relates to https://trello.com/c/otHVWblH/130-ceph-daemon-make-check-integration . Can we also use something like `src/python-common/src/deployment/ceph_daemon.py` here in the RPM?

Reply:
I can not say much about `src/python-common`. But given that `ceph-daemon` is completely independent (and my understanding is that it will be that way) I think having a extra dir is cleaner.
For unit testing I would do:
- create a tox.ini
- create a tests/ dir with the usual `unittest.Testcase` classes
- use `imp.load_module()` to load the `ceph-daemon` file and be able to test the available functions

Reply:
I can add a basic test and the structure in extra commits if you want to see that before the decision to merge or not merge.

Reply:
> I can add a basic test and the structure in extra commits if you want to see that before the decision to merge or not merge.

Done

Reply:
> I can not say much about `src/python-common`. But given that `ceph-daemon` is completely independent 

actually no. We're transferring ceph-daemon from the mgr to the remote hosts. Right now, we're doing this by depending on the ceph-daemon rpm package and then having a quirk to store the location of ceph-daemon somehere. Much easier would be to just put ceph-daemon into python-common and just use `ceph_demon.__file` when transferring it. 

Secondly, they are not really independent, but rather two components of the whole thing. At least for now.

Reply:
on the other hand, I guess I'm too opinionated. Of course, we can also get away with a decent cmake integration here.

Reply:
> > I can not say much about `src/python-common`. But given that `ceph-daemon` is completely independent
> 
> actually no. We're transferring ceph-daemon from the mgr to the remote hosts. Right now, we're doing this by depending on the ceph-daemon rpm package and then having a quirk to store the location of ceph-daemon somehere. Much easier would be to just put ceph-daemon into python-common and just use `ceph_demon.__file` when transferring it.

That way you would need to have `python-common` RPM/DEB package installed to use `ceph-daemon` (for the standalone usecase).
Being able to get the full path from python is nice. OTOH, just saying it must be in `$PATH` to find it for copying is imo also ok.

> Secondly, they are not really independent, but rather two components of the whole thing. At least for now.

What I meant with *independent* is, that `ceph-daemon` does not have any runtime requirements beside pythons stdlib. I don't need `python-common` to execute `ceph-daemon`.

Reply:
It just feels so unnecessary and wasted to have a dedicated tox environment, cmake integration, etc. just for this one file.